src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -685,7 +685,13 @@ class IAggregateFunctionDataHelper : public IAggregateFunctionHelper<Derived>
     static constexpr bool DateTime64Supported = true;
 
     IAggregateFunctionDataHelper(const DataTypes & argument_types_, const Array & parameters_)
-        : IAggregateFunctionHelper<Derived>(argument_types_, parameters_) {}
+        : IAggregateFunctionHelper<Derived>(argument_types_, parameters_)
+    {
+        static_assert(
+            std::is_same_v<decltype(&IAggregateFunctionDataHelper::destroy), decltype(&Derived::destroy)> ==
+                std::is_same_v<decltype(&IAggregateFunctionDataHelper::hasTrivialDestructor), decltype(&Derived::hasTrivialDestructor)>,
+            "CLICKHOUSE AGGREGATIONS: You need to override BOTH destroy() and hasTrivialDestructor() (or NONE of them)");
```

Comment:
Makes sense. To be fair I was screaming because the compiler error isn't very clean to read:

```
/mnt/ch/ClickHouse/src/AggregateFunctions/IAggregateFunction.h:690:9: error: static_assert failed due to requirement 'std::is_same_v<void (DB::IAggregateFunctionDataHelper<DB::AggregateFunctionArgMinMaxData<DB::SingleValueDataFixed<char8_t>, DB::AggregateFunctionMinData<DB::SingleValueDataString>>, DB::AggregateFunctionArgMinMax<DB::AggregateFunctionArgMinMaxData<DB::SingleValueDataFixed<char8_t>, DB::AggregateFunctionMinData<DB::SingleValueDataString>>>>::*)(char *) const noexcept, void (DB::AggregateFunctionArgMinMax<DB::AggregateFunctionArgMinMaxData<DB::SingleValueDataFixed<char8_t>, DB::AggregateFunctionMinData<DB::SingleValueDataString>>>::*)(char *) const noexcept> == std::is_same_v<bool (DB::IAggregateFunctionDataHelper<DB::AggregateFunctionArgMinMaxData<DB::SingleValueDataFixed<char8_t>, DB::AggregateFunctionMinData<DB::SingleValueDataString>>, DB::AggregateFunctionArgMinMax<DB::AggregateFunctionArgMinMaxData<DB::SingleValueDataFixed<char8_t>, DB::AggregateFunctionMinData<DB::SingleValueDataString>>>>::*)() const, bool (DB::IAggregateFunctionDataHelper<DB::AggregateFunctionArgMinMaxData<DB::SingleValueDataFixed<char8_t>, DB::AggregateFunctionMinData<DB::SingleValueDataString>>, DB::AggregateFunctionArgMinMax<DB::AggregateFunctionArgMinMaxData<DB::SingleValueDataFixed<char8_t>, DB::AggregateFunctionMinData<DB::SingleValueDataString>>>>::*)() const>' "CLICKHOUSE AGGREGATIONS: You need to override BOTH destroy() and hasTrivialDestructor() (or NONE of them)"
```

Reply:
Changed to something cleaner:
```
/mnt/ch/ClickHouse/src/AggregateFunctions/IAggregateFunction.h:695:9: error: static_assert failed due to requirement 'declares_destroy_and_hasTrivialDestructor' "destroy() and hasTrivialDestructor() methods of an aggregate function must be either both overridden or not"
        static_assert(declares_destroy_and_hasTrivialDestructor,
```